### PR TITLE
infra: allow file exclusions in diff report generation

### PR DESCRIPTION
**Pull request by nmancus1**
Related to discussion at https://github.com/checkstyle/contribution/pull/527#issuecomment-969376283 , allowing file exclusions in diff report generation will permit us to run check regression on newer openjdk's.

I do not think that github actions will use the proposed changes in this PR to execute a diff report created by a comment in this PR, so we will need to merge this, then try https://github.com/checkstyle/checkstyle/pull/10911 again to show that exclusions work. I will also update #10911 to use openjdk16.

### CODE REVIEW

**Comment by nmancus1** [comment]
@rnveach ping

File: .github/workflows/diff_report.yml
Comment (nmancus1) [inline .github/workflows/diff_report.yml:180]: I used the long option here, alternative is `-g` which is very cryptic to me.

**Comment by rnveach** [comment]
I thought we were only talking about the OpenJDK run, not for all report generation.
Don't you need to hook this into your contribution PR to ensure we can do the full run now?

**Comment by nmancus1** [comment]
> Don't you need to hook this into your contribution PR to ensure we can do the full run now?

As far as I know, there is no way to test this action without merging this PR; github actions does not run using PR branch, this could have security implications.

**Comment by nmancus1** [comment]
We should remove excluded files from https://github.com/checkstyle/contribution/blob/master/checkstyle-tester/projects-to-test-on-for-github-action.properties before merging this, since we have been ignoring them up until now

File: .github/workflows/diff_report.yml
Comment (rnveach) [inline .github/workflows/diff_report.yml:180]: > I thought we were only talking about the OpenJDK run, not for all report generation.

This is my only concern is that we are enabling this for all projects. We should re-look at our exlcude list and see if there is anything we want to remove.

Comment (rnveach) [reply .github/workflows/diff_report.yml:180]: @romani Please confirm you are ok with this.

Comment (romani) [reply .github/workflows/diff_report.yml:180]: it took me awhile to meditate on this...

having no excludes were ok for us to see easily that parsing is of files changed and here is diff. If somebody changing ANTRL it become useful. We do not change ANTLR that often. So probably should keep this in mind by maintainers, and let all other 95% of updates in code base to reuse less confusing environment for execution of openjdk sources.

I am ok to allowExcludes, I think it will make process more easy for outside contributors.
Excluded files because they are bad in complication is good. If we exclude some files due to new syntax, we should keep link "until .... " and our link check will catch forgotten exclude. But still it happens so rare.

**Review by rnveach — APPROVED** [review]
> We should re-look at our exclude list and see if there is anything we want to remove.

**Review by romani — APPROVED** [review]

**Comment by romani** [comment]
>We should re-look at our exclude list and see if there is anything we want to remove.

@nmancus1 , please add this to your PR for openjdk config file creation

**Pull request merged**
